pymongo/cursor.py

```diff
@@ -291,14 +284,20 @@ def _clone_base(self, session):
     def __die(self, synchronous=False):
         """Closes this cursor.
         """
-        already_killed = self.__killed
+        # the __init__ might not have run completely here (or at all)
```

Comment:
I think we can just do:
```python
try:
    already_killed = self.__killed
except AttributeError:
    # __init__ failed to run because of an invalid argument.
    return
```
We know that `__killed`, `__id`, `__exhaust`, and `__exhaust_mgr` will always be defined if `__init__` actually runs.

Reply:
yes much better, thanks! I have amended the commit to use this approach